app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataMiddleware.kt

```diff
@@ -29,6 +30,10 @@ class HistoryMetadataMiddleware(
 
     private val logger = Logger("HistoryMetadataMiddleware")
 
+    // Tracks whether a page load is in progress that was triggered directly by the app
+    // e.g. via the toolbar as opposed to via web content.
+    private var directLoadTriggered: Boolean = false
```

Comment:
I will file a follow-up in A-C to come up with something better in the long run.

Reply:
https://github.com/mozilla-mobile/android-components/issues/11034